### PR TITLE
Add error handling for other http codes

### PR DESCRIPTION
**Pull request by trondaa**

### CODE REVIEW

File: service.py [hunk 1]
```diff
@@ -70,12 +70,13 @@ def org_user(org):
                         logger.info(f"Organization invitation sent to username: {invi_username}")
                     if decoded_data.get('state') == "active":
                         logger.info(f"User with username: {invi_username}, already part of organization")
-                    
-                if invi_response.status_code == 422 or invi_response.status_code == 403:
+                elif invi_response.status_code == 422 or invi_response.status_code == 403:
                     logger.warning(f"Organization invitation could not be sent to username: {invi_username}")
                     logger.warning(f"Failed with error code: {invi_response.status_code}")
-
+                else:
```
Comment (meyhane) [inline service.py:76]: as an example - status 201 would fall here as well as 404.
logging based on `invi_response.ok` would eliminate conflicting log messages at least.

**Review by meyhane — CHANGES_REQUESTED** [review]

File: service.py (hunk 1 above)
Comment (trondaa) [reply service.py:76]: Changed it now to also handle ok, but unexpected return codes

File: service.py [hunk 2]
```diff
@@ -64,18 +64,23 @@ def org_user(org):
...
+                            logger.info(f"Organization invitation sent to username: {invi_username}")
+                        if decoded_data.get('state') == "active":
+                            logger.info(f"User with username: {invi_username}, already part of organization")
+                    else:
+                        logger.warning(f"User with username: {invi_username} invited, but unexpected response with status code: {invi_response.status_code}")
+                else:
+                    if invi_response.status_code == 422 or invi_response.status_code == 403:
+                        logger.warning(f"Organization invitation could not be sent to username: {invi_username}")
+                        logger.warning(f"Failed with error code: {invi_response.status_code}")
+                    else:
```
Comment (meyhane) [inline service.py:80]: Approving with a suggestion: 'If' and 'else' block here can be unified without checking any specific code. As long as it is non-ok response, it doesn't matter what code it is.

**Review by meyhane — APPROVED** [review]
Approved with suggestions

File: service.py (hunk 2 above)
Comment (trondaa) [reply service.py:80]: Guessing the original developer had some intention behind it so I thought I'll leave it as is

**Pull request merged**